plugins/shell_filter_file_contents/__init__.py

```diff
@@ -0,0 +1,25 @@
+#Pipe contents of each exported file through FILTER_CONTENTS <file-path> <hg-hash> <is-binary>"
+import subprocess
+import shlex
+
+def build_filter(args):
+    return Filter(args)
+
+class Filter:
+    def __init__(self, args):
+        self.filter_contents = shlex.split(args)
+
+    def file_data_filter(self,file_data):
+        d = file_data['data']
+        file_ctx = file_data['file_ctx']
+        filter_cmd = filter_contents + [filename, node.hex(file_ctx.filenode()), '1' if file_ctx.isbinary() else '0']

```

Comment:
Have you tried this?

```
Traceback (most recent call last):
  File "/home/frej/src/fast-export/hg-fast-export.py", line 586, in <module>
    plugins=plugins_dict))
  File "/home/frej/src/fast-export/hg-fast-export.py", line 452, in hg2git
    plugins)
  File "/home/frej/src/fast-export/hg-fast-export.py", line 271, in export_commit
    export_file_contents(ctx,man,added,hgtags,fn_encoding,plugins)
  File "/home/frej/src/fast-export/hg-fast-export.py", line 145, in export_file_contents
    filter(file_data)
  File "/home/frej/src/fast-export/pluginloader/../plugins/shell_filter_file_contents/__init__.py", line 15, in file_data_filter
    filter_cmd = filter_contents + [filename, node.hex(file_ctx.filenode()), '1' if file_ctx.isbinary() else '0']
NameError: global name 'filter_contents' is not defined
```

`filename` and `node` also look suspicious.

Reply:
My mistake  - I think I may have had it working in the past, and lost it in the various reworks, but I can't be certain. I have fixed this (and you were right, filename and node are also suspicious, as was sys) and verified that all three plugins are currently working as intended.